ergonomics/ide.ergonomics/src/org/netbeans/modules/ide/ergonomics/fod/ConfigurationPanel.java

```diff
@@ -190,6 +180,40 @@ public void setInfo(FeatureInfo info, String displayName, Collection<UpdateEleme
             org.openide.awt.Mnemonics.setLocalizedText(downloadButton, btnDownloadMsg);
         }
     }
+    
+    /**
+     * Collect extra modules texts
+     */
+    protected List<String> collectExtraModulesTextsFromFeatures(Collection<FeatureInfo> features, boolean required) {
+        List<String> descriptionsList = new ArrayList<>();
+        for (FeatureInfo fi : features) {
+            String s = required ? fi.getExtraModulesRequiredText(): fi.getExtraModulesRecommendedText();
+            if (!descriptionsList.contains(s)) {
+                descriptionsList.add(s);
+            }
+        }
+        return descriptionsList;
+    } 
+    
+    /**
+     * Generate download message from extra modules texts
+     * @param extraModulesTexts
+     * @return String Text to set in download label
+     */
+    protected String generateDownloadMessageFromExtraModulesTexts(List<String> extraModulesTexts) {
+        StringBuilder sbDownload = new StringBuilder();
```

Comment:
A suggestion: these two functions (`collectExtraModulesTextsFromFeatures` and `generateDownloadMessageFromExtraModulesTexts`) are called in a sequence: you could merge it, producing the String for the label - and you could eventually print the *required* texts in Bold into the string ?

Reply:
I could merge both functions, but I created these to allow create tests for this part of the code. 
Is not the best solution but is too hard to test the ConfigurationPanel class due his dependencies.

Reply:
OK, it was just an idea; let's see how the entire message list will be readable